### PR TITLE
composter fix

### PR DESCRIPTION
**Pull request by hannibal002**
This looks good, right?

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt [hunk 1]
```diff
@@ -64,6 +66,8 @@ class ComposterOverlay {
         var inInventory = false
     }
 
+    var garden: JsonObject? = null
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt]: Me when raw json object

File: src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt [hunk 2]
```diff
@@ -330,6 +339,12 @@ class ComposterOverlay {
         newList.addAsSingletonList("")
     }
 
+    private val mutableMap: MutableMap<String, Double>
+        get() {
+            val baseValues = mutableMapOf<String, Double>()
+            return baseValues
+        }
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt]: This seems like an entirely pointless variable.

File: src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt [hunk 3]
```diff
@@ -379,22 +394,32 @@ class ComposterOverlay {
...
-            }
-            organicMatterFactors = updateOrganicMatterFactors(baseValues)
+    private fun updateOrganicMatterFactors() {
+        try {
+            garden?.let {
+                val baseValues = mutableMapOf<String, Double>()
+                for ((name, value) in it["organic_matter"].asJsonObject.entrySet()) {
+                    baseValues[name] = value.asDouble
+                }
+                organicMatterFactors = updateOrganicMatterFactors(baseValues)
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt]: These can probably be inlined: 
```suggestion
                organicMatterFactors = updateOrganicMatterFactors(it["organic_matter"].asJsonObject.entrySet().associate { it.key to it.value.asDouble })
```

**Review by lineargraph — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt (hunk 3 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt]: I think i have implemented that correctly

File: src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt]: Me when effort for java objects

File: src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt (hunk 2 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt]: deleted.

File: src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt]: done

File: src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt (hunk 3 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt]: done

**Pull request merged**